scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -308,7 +308,14 @@ def get_all_gem_infos(project_path: pathlib.Path or None) -> list:
     # it's easier to determine which gems are engine gems here rather than in c++
     # because the project might be using a different engine than the one Project Manager is
     # running out of
-    engine_path = manifest.get_project_engine_path(project_path=project_path) if project_path else manifest.get_this_engine_path() 
+
+    # Attempt to get the engine path from the project path if possible. If not, then default
+    # to attempting to get engine path from the registered '--this-engine' value from the manifest
+    engine_path = None
+    if project_path:
+        engine_path = manifest.get_project_engine_path(project_path=project_path)
+    if not engine_path:
+        engine_path = manifest.get_this_engine_path()
```

Comment:
You can restrict this logic to only occur for templates by detecting if the project path is underneath a template directory via using the [utils.find_ancestor_dir_containing_file](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/utils.py#L497C5-L507) method.

```python
is_project_template = utils.find_ancestor_dir_containing_file('template.json', project_path) != None;
```